pkg/microservice/aslan/core/workflow/handler/plugin_repo.go

```diff
@@ -29,3 +36,38 @@ func ListPluginTemplates(c *gin.Context) {
 
 	ctx.Resp, ctx.Err = workflow.ListPluginTemplates(ctx.Logger)
 }
+
+func ListUnofficalPluginRepositories(c *gin.Context) {
+	ctx := internalhandler.NewContext(c)
+	defer func() { internalhandler.JSONResponse(c, ctx) }()
+
+	ctx.Resp, ctx.Err = workflow.ListUnofficalPluginRepositories(ctx.Logger)
+}
+
+func DeletePluginRepo(c *gin.Context) {
+	ctx := internalhandler.NewContext(c)
+	defer func() { internalhandler.JSONResponse(c, ctx) }()
+
+	ctx.Err = workflow.DeletePluginRepo(c.Param("id"), ctx.Logger)
+}
+
+func UpsertUserPluginRepository(c *gin.Context) {
+	ctx := internalhandler.NewContext(c)
+	defer func() { internalhandler.JSONResponse(c, ctx) }()
+
+	req := &commonmodels.PluginRepo{}
+	data, err := c.GetRawData()
+	if err != nil {
+		log.Errorf("UpsertUserPluginRepository err: %s", err)
+	}
+	if err = json.Unmarshal(data, req); err != nil {
+		log.Errorf("UpsertUserPluginRepository unmarshal json err: %s", err)
+	}
+	c.Request.Body = ioutil.NopCloser(bytes.NewBuffer(data))
+
```

Comment:
if we don't add operation log for this api, there is no need to unmarshal the request body twice.